src/privileged/blurts/FirefoxMonitor.jsm

```diff
@@ -0,0 +1,652 @@
+
+  if (panel.hidden) {
+    showPanel(browser, ui, "breach-alerts-survey");
+    return;
+  }
+
+  panel.addEventListener("popuphidden", () => {
+    showPanel(browser, ui, "breach-alerts-survey");
+  }, { once: true });
+}
+
+function showPanel(browser, ui, notificationId) {
+  let doc = browser.ownerDocument;
+  let panel = doc.defaultView.PopupNotifications.panel;
+
+  let populatePanel = (event) => {
+    if (event !== "shown") {
+      return;
+    }
+    let n = doc.getElementById(notificationId + "-notification");
+    let body = doc.getAnonymousElementByAttribute(n, "class", "popup-notification-body");
+    let box = body.querySelector(".blurtsbox");
+    if (box) {
+      box.remove();
+    }
+    box = ui.box;
+    box.setAttribute("class", "blurtsbox");
+    box.setAttribute("style", "font-size: 110%");
+    body.appendChild(box);
+    let icon = doc.getAnonymousElementByAttribute(n, "class", "popup-notification-icon");
+    if (icon) {
+      icon.remove();
+    }
+  };
+
+  doc.defaultView.PopupNotifications.show(
+    browser, notificationId, "",
+    null, ui.primaryAction, ui.secondaryActions, {persistent: true, eventCallback: populatePanel});
+}
+
+const XUL_NS = "http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul";
+let UIFactory = [
+  function(browser, doc, host, site) {
+    return {
+      get box() {
+        let box = doc.createElementNS(XUL_NS, "vbox");
+        let elt = doc.createElementNS(XUL_NS, "hbox");
+        elt.setAttribute("style", "margin-bottom: 1rem;");
+        elt.setAttribute("pack", "center");
+        let imgElt = doc.createElementNS(XUL_NS, "image");
+        imgElt.setAttribute("src", imageDataURIs.iconLaptopRisk);
+        imgElt.setAttribute("style", "width: 148px;");
+        elt.appendChild(imgElt);
+        box.appendChild(elt);
+        elt = doc.createElementNS(XUL_NS, "description");
+        elt.setAttribute("anonid", "maindesc");
+        elt.appendChild(doc.createTextNode("Have an account on this website?\nIt may be at risk."));
+        elt.setAttribute("style", "font-size: 150%; text-align: center; white-space: pre; margin-bottom: 1rem;");
+        box.appendChild(elt);
+        elt = doc.createElementNS(XUL_NS, "description");
+        elt.appendChild(doc.createTextNode(site.Title + " was reported to Firefox Monitor, a service that collects information about data breaches and other ways hackers can steal your information.\n\nFind out if your account is at risk and learn how to protect yourself at the Firefox Monitor website."));
+        elt.setAttribute("style", "text-align: center; white-space: pre-wrap;");
+        box.appendChild(elt);
+        return box;
+      },
+      primaryAction: {
+        label: "Go to Firefox Monitor",
+        accessKey: "f",
+        callback: () => {
+          FirefoxMonitor.notifyTelemetryListeners(`variant_1_submit`);
+          doc.defaultView.openUILinkIn("http://fx-breach-alerts.herokuapp.com/?breach=" + host, "tab");
```

Comment:
Probably not a big deal (and I think we have an open issue for updating this URL to the prod endpoint anyways), but note that a few of these urls are HTTP:// and not HTTPS://.

Also can't remember if mentioned elsewhere, or if we need to URLEncode the `host` since we're passing it as a query string.

Reply:
Also, I don't think this should be `host`, per #24.

For example, the URL that we want (at least for the server), is https://fx-breach-alerts.herokuapp.com/?breach=Adobe.
This actually gets a bit interesting in the case of domains with multiple breaches (ie: bell.ca).

https://fx-breach-alerts.herokuapp.com/?breach=Bell
https://fx-breach-alerts.herokuapp.com/?breach=Bell2017

So not sure what field in breaches.json we should pull from to get that slug. It _almost_ looks like the `LogoFilename` minus the file extension:

```json
{
  "Domain": "bell.ca",
  "Title": "Bell (2017 breach)",
  "BreachDate": "2017-05-15",
  "PwnCount": 2231256,
  "DataClasses": ["Email addresses","Geographic locations","IP addresses","Job titles","Names","Passwords","Phone numbers","Spoken languages","Survey results","Usernames"],
  "LogoFilename": "Bell2017.svg"
}
```